### DISCUSSION TITLE
How to add exceptions for cleanup or view outdated packages？

### DISCUSSION THREAD

**Discussion by eidswe** [ScoopInstaller/Scoop#5372] [General]
The problem is that some packages need to keep a certain old version and the latest version, directly using scoop cleanup * will uninstall these packages.

**Comment by rashil2000** [comment]
You can use the PowerShell pipeline to do this:

```pwsh
scoop list | ? name -notin app1,app2,app3 | % { scoop cleanup $_.name }
```

**Reply by eidswe** [reply]
Thank you! Also is there a way to check for obsolete packages without immediately removing them?

**Comment by rashil2000** [comment]
`scoop status`
